src/Extension/Lesson.cs

```diff
@@ -9,37 +9,21 @@ namespace Extension
 {
     public class Lesson
     {
-        public ChallengeController ChallengeController { get; private set; } = new ChallengeController();
-        public Challenge CurrentChallenge
-        {
-            get
-            {
-                return ChallengeController.CurrentChallenge;
-            }
-            set
-            {
-                ChallengeController.CurrentChallenge = value;
-            }
-        }
+        public Challenge CurrentChallenge { get; private set; }
 
         public Lesson()
         {
 
         }
 
-        public void AddChallenge(Challenge challenge)
-        {
-            ChallengeController.AddChallenge(challenge);
-            challenge.Lesson = this;
-        }
-
-// todo: remove pragma
+        // todo: remove pragma
 #pragma warning disable 1998
         public async Task StartChallengeAsync(Challenge challenge)
 #pragma warning restore 1998
         {
             CurrentChallenge = challenge;
-            // todo: await someexternalendpoint.StartChallenge
+            CurrentChallenge.Focus();
```

Comment:
This looks like spooky action at a distance. Whatever class does the work of "focus" should be called directly.

Reply:
updated